src/places.py

```diff
@@ -384,3 +420,37 @@ def _on_encrypted_eject_finished(self, encrypted, success):
             self._on_device_removed(None, encrypted)
         else:
             self.emit("failed", None)
+
+    def canBookmark(self, path):
```

Comment:
Then, all this logic could go into PortfolioBookmarks and PortfolioPlaces would interact with it through signals like it does with PortfolioDevices.